runtime/alloc/collect.c

```diff
@@ -111,6 +112,78 @@ static void migrate_string_buffer(stringbuffer** bufferPtr) {
   *bufferPtr = *(stringbuffer **)(buffer->contents);
 }
 
+static void migrate_mpz(mpz_ptr *mpzPtr) {
+  mpz_hdr *intgr = struct_base(mpz_hdr, i, *mpzPtr);
+  const uint64_t hdr = intgr->h.hdr;
+  bool isNotInYoungGen = hdr & NOT_YOUNG_OBJECT_BIT;
+  bool hasAged = hdr & YOUNG_AGE_BIT;
+  if (isNotInYoungGen && !(hasAged && collect_old)) {
```

Comment:
I think something is poorly named around here. Why is a mask called `YOUNG_AGE_BIT` considered when the object `isNotInYoungGen`?

Reply:
YOUNG_AGE_BIT gets set when the object survives its first young generation collection. It then remains set when the object is promoted to the old generation. We then check both YOUNG_AGE_BIT and NOT_YOUNG_OBJECT_BIT because a combination of both tells us that the object is in the old generation. If you have a better name to suggest, I'm willing to listen though.

Reply:
The `YOUNG` only refers to the role of that bit in the young generation. If it means the same thing in headers in the old generation it could have a more generic name like `AGE_BIT`. If the bit doesn't really mean the same thing, there could be multiple names for the same mask value and I expect the optimizer would remove any apparent redundancy (especially considering hdr is used from a local variable). Perhaps a macro or inline function `isInOldGen(hdr)` computing `(hdr & NOT_YOUNG_OBJECT_BIT) && (hdr & YOUNG_AGE_BIT)`,

Reply:
In general I think I agree with @bmmoore that the naming of the flags is a little confusing. I don't think though that renaming them should be part of this PR. So if that's ok with everybody, I will go ahead and merge it.

Reply:
I don't think it's too much effort to make a macro like Brandon suggested. Why don't we aim for it as part of this pr, even if we don't rename anything.

Reply:
OK that sounds fair. I will do it

Reply:
I pushed a commit that adds the discussed macro. @bmmoore please take a look again